File: packages/core/src/programmers/helpers/ICheckEntry.ts
import ts from "@typescript/native-preview";

export interface ICheckEntry {
  expected: string;
  expression: ts.Expression | null;
  conditions: ICheckEntry.ICondition[][];
}
export namespace ICheckEntry {
  export interface ICondition {
    expected: string;
    expression: ts.Expression;
  }
}

File: packages/typia/native/core/programmers/helpers/ICheckEntry.go
package helpers

import shimast "github.com/microsoft/typescript-go/shim/ast"

type ICheckEntry struct {
	Expected   string
	Expression *shimast.Node
	Conditions [][]ICheckEntry_ICondition
}

type ICheckEntry_ICondition struct {
	Expected   string
	Expression *shimast.Node
}
